crates/pipeline_manager/src/integration_test.rs

```diff
@@ -974,3 +974,112 @@ async fn quoted_columns() {
         .wait_for_pipeline_status(&id, PipelineStatus::Shutdown, config.shutdown_timeout)
         .await;
 }
+
+#[actix_web::test]
+#[serial]
+async fn primary_keys() {
+    let config = setup().await;
+    let id = deploy_pipeline_without_connectors(
+        &config,
+        r#"create table t1(id bigint not null, s varchar not null, primary key (id))"#,
+    )
+    .await;
+
+    // Start the pipeline
+    let resp = config
+        .post_no_body(format!("/v0/pipelines/{}/start", id))
+        .await;
+    assert_eq!(resp.status(), StatusCode::ACCEPTED);
+    config
+        .wait_for_pipeline_status(&id, PipelineStatus::Running, Duration::from_millis(1_000))
+        .await;
+
+    // Push some data using default json config.
+    let req = config
+        .post_json(
+            format!(
+                "/v0/pipelines/{}/ingress/T1?format=json&update_format=insert_delete",
+                id
+            ),
+            r#"{"insert":{"id":1, "s": "1"}}
+{"insert":{"id":2, "s": "2"}}"#
```

Comment:
We could use the json macro + `.to_string()` for these json literals, could make things nicer (especially the following json literals)